src/neo/Consensus/ConsensusContext.cs

```diff
@@ -405,6 +423,38 @@ public void Reset(byte viewNumber)
             TransactionHashes = null;
             PreparationPayloads = new ConsensusPayload[Validators.Length];
             if (MyIndex >= 0) LastSeenMessage[MyIndex] = (int)Block.Index;
+
+            if (resetFutures)
+                ResetFuturePayloads();
+        }
+
+        public void ResetFuturePayloads()
+        {
+            FutureCommitPayloads = new ConsensusPayload[Validators.Length];
+            FuturePreparationPayloads = new ConsensusPayload[Validators.Length];
+            FutureChangeViewPayloads = new ConsensusPayload[Validators.Length];
+            FutureRecoveryPayloads = new ConsensusPayload[Validators.Length];
+        }
+
+        public bool HasFuturePayloads()
+        {
+            if (FuturePreparationPayloads != null)
```

Comment:
Can we have a flag instead of search in all of them?

Reply:
I do not know how to design this flag efficiently right now, because sometimes we use a `FuturePayload` and we would need to make a check similar like this to see if the flag will be updated.

Reply:
Yes, we don't need a flag, is more like a counter. I think that it should be faster.

Reply:
That is true, a counter is a good idea!

Reply:
Done, @shargon.